src/webui/api/torrentscontroller.cpp

```diff
@@ -782,7 +782,9 @@ void TorrentsController::addTrackersAction()
     if (!torrent)
         throw APIError(APIErrorType::NotFound);
 
-    const QVector<BitTorrent::TrackerEntry> entries = BitTorrent::parseTrackerEntries(params()[u"urls"_s]);
+    const QVector<BitTorrent::TrackerEntry> trackers = torrent->trackers();
+    int maxTier = (trackers.size() > 0 ? trackers.back().tier : -1);  // get maximum tier of current torrent
+    const QVector<BitTorrent::TrackerEntry> entries = BitTorrent::parseTrackerEntries(params()[u"urls"_s], maxTier + 1);
```

Comment:
I don't think you should change this API. What was the reason?

Reply:
I believe we also need to take care about webui as they share same tracker parsing functions?